### ISSUE TITLE
CMake Error at CMakeLists.txt:713 (execute_process) "Child return code: 3"

### ISSUE THREAD

**Issue by MartinX3** [moneymanagerex/moneymanagerex#6693]
MMEX version:
 - [x] 1.8.0

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [ ] Windows
 - [ ] Mac OSX
 - [x] Linux 

# Description of the bug

Error while compilation

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:
Compile tag 1.8.0

Expected result:
Compiled release

Actual result:
Error

# Additional information
Compile script: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=moneymanagerex

Log:
```
==> Beginne build()...
-- Build type: Stable
-- The C compiler identification is GNU 13.2.1
-- The CXX compiler identification is GNU 13.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib/ccache/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib/ccache/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.44.0")
-- Found compile cache tool: /usr/bin/ccache (found version "4.9.1")
-- Found wxWidgets: -pthread;-lwx_gtk3u_core-3.2;-lwx_gtk3u_qa-3.2;-lwx_gtk3u_html-3.2;-lwx_baseu_xml-3.2;-lwx_gtk3u_aui-3.2;-lwx_gtk3u_stc-3.2;-lwx_gtk3u_webview-3.2;-lwx_baseu-3.2 (found suitable version "3.2.4", minimum required is "2.9.2")
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.1.1")
-- Checking for one of the modules 'libcurl'
-- Found CURL: /usr/lib/libcurl.so (found version "8.7.1")
-- Module support is disabled.
-- Version: 8.1.1
-- Build type: None
-- CXX_STANDARD: 11
-- Performing Test has_std_11_flag
-- Performing Test has_std_11_flag - Success
-- Performing Test has_std_0x_flag
-- Performing Test has_std_0x_flag - Success
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS - Success
-- Performing Test FMT_HAS_VARIANT
-- Performing Test FMT_HAS_VARIANT - Success
-- Required features: cxx_variadic_templates
-- Looking for strtod_l
-- Looking for strtod_l - not found
-- RapidJSON found. Headers: /usr/include
-- Found RapidJSON: /usr/include
-- Found Lua: /usr/lib/liblua5.4.so;/usr/lib/libm.so (found version "5.4.6")
-- Looking for dlopen in dl
-- Looking for dlopen in dl - found
-- Found Gettext: /usr/bin/msgmerge (found version "0.22.4")
-- Performing Test Wduplicated-cond
-- Performing Test Wduplicated-cond - Success
-- Performing Test Wduplicated-branches
-- Performing Test Wduplicated-branches - Success
-- Performing Test Wnull-dereference
-- Performing Test Wnull-dereference - Success
-- Performing Test Wlogical-op
-- Performing Test Wlogical-op - Success
-- Performing Test Wuseless-cast
-- Performing Test Wuseless-cast - Success
-- AppStream version AppStream-Version: 1.0.3
I: org.moneymanagerex.MMEX:106: developer-name-tag-deprecated

✘ Validierung fehlgeschlagen: Informationen: 1, Pedantisch: 1
CMake Error at CMakeLists.txt:713 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 3"



-- Configuring incomplete, errors occurred!
==> FEHLER: Ein Fehler geschah in build().
    Breche ab...
Fehler: ‚moneymanagerex-1.8.0-2‘ konnte nicht erstellt werden: 
Fehler: Pakete konnten nicht erstellt werden: moneymanagerex-1.8.0-2
```

**Comment by n-stein** [comment]
Appstream 1.0.3 (which was just released 5 days ago) deprecated the developer_name tag in metainfo. This line needs to be removed:

https://github.com/moneymanagerex/moneymanagerex/blob/c961d79554337c8677a601852a8a439c2e899613/resources/dist/linux/share/metainfo/org.moneymanagerex.MMEX.metainfo.xml.in#L96

In order to build 1.8.0 you need to either remove that line from `resources/dist/linux/share/metainfo/org.moneymanagerex.MMEX.metainfo.xml.in` prior to running `cmake .` or downgrade your Appstream version to 1.0.2

**Comment by ovari** [comment]
Will the information (shown in image below) be populated from somewhere else? Thank you

![image](https://github.com/moneymanagerex/moneymanagerex/assets/17465872/7febb7da-8622-4a4b-ac01-6987be928692)

**Comment by MartinX3** [comment]
@ovari please look at the code.
It's the line directly under the deprecated `developer_name`.

@joshua-stone just forget to remove the deprecated line after adding the new replacement.
https://github.com/moneymanagerex/moneymanagerex/commit/3c3f15a5e514203f2c7a0b37c609b4c572c54b6d

**Comment by joshua-stone** [comment]
> @ovari please look at the code. It's the line directly under the deprecated `developer_name`.
> 
> @joshua-stone just forget to remove the deprecated line after adding the new replacement. See [3c3f15a](https://github.com/moneymanagerex/moneymanagerex/commit/3c3f15a5e514203f2c7a0b37c609b4c572c54b6d)

That line was deliberately left in place because the appstreamcli-based file validation in Flathub's CI differed from what was provided in Flatpak SDKs since the SDKs use an older appstreamcli version:

```
$ flatpak run --command=appstreamcli org.freedesktop.Sdk//23.08 --version
AppStream version: 0.16.4
$ flatpak run --command=appstreamcli org.gnome.Sdk//45 --version
AppStream version: 0.16.4
$ flatpak run --command=appstreamcli org.gnome.Sdk//46 --version
AppStream version: 0.16.4
```

https://github.com/ximion/appstream/blob/0f83f572c89f936ddb3c5bf1a8fd85e8c3976a68/NEWS#L222

This resulted in a bug where publishing of MMEX 1.7.0 to Flathub failed, requiring an extra post-install step to make sure the appstream validation could pass on Flathub:

https://github.com/flathub/org.moneymanagerex.MMEX/pull/15
https://github.com/flathub/org.moneymanagerex.MMEX/pull/16

Some time later this bug was fixed, which meant that the extra step could be removed for MMEX 1.8.0:

https://github.com/flathub/org.moneymanagerex.MMEX/pull/17

**Comment by MartinX3** [comment]
@joshua-stone 
Thank you.
Another reason I need to avoid these isolated package images and use system packages instead.

**Comment by joshua-stone** [comment]
> @joshua-stone Thank you. Another reason I need to avoid these isolated package images and use system packages instead.

The core issue wasn't whether the packages were isolated or not, rather there was a mismatch in environments at the build step time and the deployment step, .i.e., one step was looking for the `<developer>` field while the other was looking for `<developer_name>` field. The bug appears to have been fixed on Flathub, so we can move forward with removing `<developer_name>`.

Isolating dependencies with flatpak SDKs instead of building with system packages was chosen for the upstream Linux build because MMEX required very specific versions of dependencies, and many of them were not guaranteed to be available as pre-built distro packages. Some of those dependencies included (but not limited to):

- wxWidgets
- libsoup-2.4
- webkitgtk

https://github.com/flathub/org.moneymanagerex.MMEX/pull/11

Projects compiling against a specific version of wxWidgets is a common occurrence for this reason:

https://github.com/search?q=org%3Aflathub+https%3A%2F%2Fgithub.com%2FwxWidgets%2FwxWidgets&type=code

Libraries may have to be bundled or statically linked if you have to use a library that isn't already packaged for a distro or has the right version. There might even be some patching required to ensure specific features are enabled and are working consistently. Flatpak helps to solve this problem.

https://tesk.page/2023/06/04/response-to-developers-are-lazy-thus-flatpak/#so-what-about-traditional-distributions

**Comment by MartinX3** [comment]
@n-stein 
I tried your patch.
Sadly I got the same result.

```
I: org.moneymanagerex.MMEX:106: developer-name-tag-deprecated

✘ Validierung fehlgeschlagen: Informationen: 1, Pedantisch: 1
CMake Error at CMakeLists.txt:713 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 3"
```

**Comment by n-stein** [comment]
The call to `appstreamcli news-to-metainfo` is automatically adding it back in. I logged a bug for it https://github.com/ximion/appstream/issues/631

In the meantime, I've pushed a patch for CMakeLists.txt that removes the bad tag before validation.

**Comment by n-stein** [comment]
Matthias just fixed the bug, so appstream 1.0.4 shouldn't have any issue building mmex <= 1.8.0

**Comment by PMaff** [comment]
I have the same issue with version 1.8.0 and
AppStream version 1.0.3  on Tumbleweed openSUSE.20240517

A build of 1.8.0 in a virtual Suse Tumbleweed with
AppStream version 1.0.2  on Tumbleweed openSUSE.20240512 works.

This also works now, thx @n-stein :
Money Manager Ex
• Version: 1.8.1-Beta.1 64-bit
• Erstellt am: Mai 22 2024 21:47:13
• Datenbankversion: 19 (aes128cbc)
• Git-Eintrag: 83ad0331f (2024-05-21)
• Git Zweig: master
MMEX verwendet folgene Unterstützungsprodukte:
• wxWidgets 3.2.4 (wxGTK 3.24)
• wxSQLite3 4.9.1 (SQLite 3.40.0)
• RapidJSON 1.1.0
• Lua 5.4.6
• libcurl/8.7.1 OpenSSL/3.1.4 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libpsl/0.21.5 libssh/0.10.6/openssl/zlib nghttp2/1.61.0 OpenLDAP/2.6.7
• gettext 0.22.5
• apexcharts.js
erstellt mit:
• CMake 3.29.3
• GNU Make 4.4.1
• GCC 13.2.1 20240509 [revision b7a2697733d19a093cbdd0e200ffce069a4bc812]
Läuft auf:
• openSUSE.20240517
• openSUSE Tumbleweed ""
• Linux 6.8.9-1-default x86_64
• XFCE (X11)
• Systemsprache: en_GB.utf8
• Sprache der Benutzeroberfläche: de_DE.UTF-8
• 0x0 0-bit 96x96ppi

Small German error :
MMEX verwendet folgene Unterstützungsprodukte
->
MMEX verwendet folgende Unterstützungsprodukte

0x0 0-bit 96x96ppi also seems a bit strange, but I cannot see an error from that up to now.
